src/script/cpp_api/s_entity.h

```diff
@@ -48,4 +49,8 @@ class ScriptApiEntity
 private:
 	bool luaentity_run_simple_callback(u16 id, ServerActiveObject *sao,
 		const char *field);
+
+	void logDeprecationForExistingProperties(lua_State *L, int index, const std::string &name);
+
+	std::unordered_set<std::string> deprecation_warned_init_properties;
```

Comment:
Documentation with explanation why this is needed would be nice, something like:
```suggestion
	/** Stores names of entities that already caused a deprecation warning due to
	 * properties being outside of initial_properties. If an entity's name is in here,
	 * it won't cause any more of those deprecation warnings. */
	std::unordered_set<std::string> deprecation_warned_init_properties;
```